include/swift/AST/Decl.h

```diff
@@ -4821,6 +4825,8 @@ class VarDecl : public AbstractStorageDecl {
     return hasName() ? getBaseIdentifier().str() : "_";
   }
 
+  int getFieldIndex() const;
```

Comment:
I feel like this should probably have a slightly scarier name to convey it might entail a non-trivial amount of work (such as synthesising backing storage vars) – something like `getABIStoredPropertyIndex` perhaps? Or even just `getABIFieldIndex`? I'd be interested to hear what others think.